frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -1245,16 +1245,6 @@ def matmul(self, resulting_type: ConversionType, x: Conversion, y: Conversion) -
             }
             self.error(highlights)
 
-        if x.is_encrypted and y.is_encrypted:
-            highlights = {
-                x.origin: "lhs is encrypted",
-                y.origin: (
-                    "rhs is encrypted" if x.origin is not y.origin else "operand is encrypted"
-                ),
-                self.converting: "but encrypted-encrypted matrix multiplications are not supported",
-            }
-            self.error(highlights)
-
         assert self.is_bit_width_compatible(resulting_type, x, y)
 
         x = self.to_signedness(x, of=resulting_type)
```

Comment:
Well, it turns out this is not enough (this is on me, sorry) :upside_down_face:

Could you apply the same patch I did in https://github.com/zama-ai/concrete/pull/394 and test it please :pray:

Reply:
I added code to promote the unsigned -> signed in the FHE datatypes of the operands involved, could you please check again?